integrations/eureka/eureka/src/main/java/io/helidon/integrations/eureka/EurekaRegistrationFeature.java

```diff
@@ -0,0 +1,1039 @@
+        // forcibly unregisters the instance. I'm not sure what the status setting accomplishes, although buried in the
+        // sediment seems to be some kind of status change event system that pertains to Eureka's peer-to-peer
+        // replication machinery that we may simply not care about here. We'll follow suit in case this sequencing turns
+        // out to be important.
+        this.up(instanceInfo, false);
+        // https://github.com/Netflix/eureka/blob/v2.0.4/eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java#L55
+        JsonObject instance = instanceInfo.getJsonObject("instance");
+        return this.cancel(client, instance.getString("app"), instance.getString("instanceId"));
+    }
+
+    // DELETE {baseUri}/v2/apps/{appName}/{id}
+    private boolean cancel(HttpClient<? extends Http1ClientRequest> client, String appName, String id) {
+        try (var response = client
+             .delete("/v2/apps/" + appName.toString() + "/" + id.toString()) // trigger NPEs if needed
+             .request()) {
+            if (response.status().family() == SUCCESSFUL) {
+                if (LOGGER.isLoggable(DEBUG)) {
+                    LOGGER.log(DEBUG,
+                               "DELETE /v2/apps/" + appName.toString() + "/" + id.toString() + ": " + response.status());
+                }
+                return true;
+            } else if (LOGGER.isLoggable(WARNING)) {
+                LOGGER.log(WARNING,
+                           "DELETE /v2/apps/" + appName.toString() + "/" + id.toString() + ": " + response.status());
+            }
+            return false;
+        }
+    }
+
+    // PUT {baseUri}/v2/apps/{appName}/{id}?status={status}&lastDirtyTimestamp={lastDirtyTimestamp}
+    //
+    // (...&overriddenstatus={someOverriddenStatus} is recognized by the Eureka server, but never sent by Eureka's
+    // registration client.)
+    private HttpClientResponse heartbeat(String appName,
+                                         String id,
+                                         String status,
+                                         Long lastDirtyTimestamp) {
+        var request = this.client // volatile read
+            .put("/v2/apps/" + appName.toString() + "/" + id.toString()) // trigger NPEs if necessary
+            .accept(APPLICATION_JSON);
+        if (status != null) {
+            request.queryParam("status", status);
+        }
+        if (lastDirtyTimestamp != null) {
+            request.queryParam("lastDirtyTimestamp", lastDirtyTimestamp.toString());
+        }
+        return request.request();
+    }
+
+    /**
+     * Marks this microservice as being {@code UP} for purposes of Eureka registration and for no other purpose.
+     *
+     * @return {@code true} if the status change was successfully recorded for eventual registration or renewal; {@code
+     * false} if no action was taken
+     */
+    public boolean markUp() {
```

Comment:
Same as below - `markDown`

Reply:
See prior comment. Let me know what to do.

Reply:
Please remove public.

Reply:
Will do; see above